clients/admin-ui/src/features/dataset/DatasetCollectionView.tsx

```diff
@@ -0,0 +1,71 @@
+import { Box, Select, Spinner } from "@fidesui/react";
+import { ChangeEvent, useEffect, useState } from "react";
+
+import { FidesKey } from "../common/fides-types";
+import { useGetDatasetByKeyQuery } from "./dataset.slice";
+import DatasetFieldsTable from "./DatasetFieldsTable";
+import { DatasetCollection } from "./types";
+
+const useDataset = (key: FidesKey) => {
+  const { data, isLoading } = useGetDatasetByKeyQuery(key);
+
+  return {
+    isLoading,
+    dataset: data,
+  };
+};
+
+interface Props {
+  fidesKey: FidesKey;
+}
+
+const DatasetCollectionView = ({ fidesKey }: Props) => {
+  const { dataset, isLoading } = useDataset(fidesKey);
+  const [activeCollection, setActiveCollection] = useState<
+    DatasetCollection | undefined
+  >();
+
+  useEffect(() => {
+    if (dataset) {
```

Comment:
If you initialize above with useState, I think you could just have the one line instead of the if/else, and then it'll update whenever dataset changes.

Reply:
yeah I see your point— I could remove the `else` even as-is! I feel a bit safer having the `else` though, so that just in case `dataset` becomes undefined (maybe a dataset of that particular `fides_key` can't be found) we don't have a stale collection hanging out.